src/Notifier.js

```diff
@@ -133,29 +133,24 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
-                        dis.dispatch({
-                            action: "notifier_enabled",
-                            value: true
-                        });
-                    }
-                });
-            }
-
-            if (!global.localStorage) return;
-            global.localStorage.setItem('notifications_enabled', 'true');
-
-            if (this.havePermission) {
+            // Case when we do not have the permission as 'granted'
+            // Attempt to get permission from user
+            global.Notification.requestPermission(function(result) {
+                if (result !== 'granted') {
+                    // The permission request was dismissed or denied
+                    return;
+                }
+
+                if (callback) callback();
```

Comment:
I think you need to update the localStorage setting before you run the callback and dispatch the `notifier_enabled` setting, otherwise things will get confused.